calicoctl/calicoctl/commands/node/diags.go

```diff
@@ -120,23 +120,30 @@ func runDiags(logDir string) error {
 	diagsTmpDir := filepath.Join(tmpDir, "diagnostics")
 
 	for _, v := range cmds {
-		err = writeDiags(v, diagsTmpDir)
-		// sometimes socket information is not collected as netstat tool
-		// is obsolete and removed in Ubuntu and other distros. so when
-		// "netstat -a -n " fails, we should use "ss -a -n" instead of it
-		if err != nil && v.cmd == "netstat -a -n" {
+		if v.cmd != "netstat -a -n" {
```

Comment:
I think this logic belongs outside of this loop. 

we generate a list of commands to run on line 83 - we should instead do something like this:

```
cmds := []diagCmd{ . . .}

if netstatExists {
  cmds = append(cmds, netstatCmd)
} else {
  cmds = append(cmds, ssCmd)
}
```

Reply:
Hi @caseydavenport 
Following your suggestion can indeed reduce some redundant code, but I just added a judgment to the previous code. If this is an optimization, append will always add elements to the end of the slice. I don't know if this is reasonable.

![image](https://github.com/projectcalico/calico/assets/112474703/24d99dfe-2c37-436e-94c9-4b497560419e)

Reply:
Yep, I think that's totally fine if the commands are run in a different order.

Reply:
@caseydavenport  please review